TerriaJS/terriajs
# TSIfy SharePanel and convert to ES6 class instead of createReactClass

### What this PR does

Fixes #6353

The purpose of this PR is to convert SharePanel.js to Typescript. Have also:
- Converted to ES6 class (including binding methods to `this` etc)
- Got rid of `IncludeStoryOption.js` and moved inside `SharePanel.ts`
- Moved `includeStoryInShare` out of `ViewState` and into local state in `SharePanel.ts`

- Also fix bug where unsharable warning panel was not showing up in SharePanel
  
### Test me
  
Go to http://ci.terria.io/tsify-sharepanel/

Click Share/Print button in top right and try to modify the advanced settings:
![image](https://user-images.githubusercontent.com/7980991/178423341-16971843-3e05-457b-b4c5-1836aa0c67d4.png)

To test warning panel:
Go to http://ci.terria.io/tsify-sharepanel/
Add a local `csv` file by drag and drop
Attempt to share the map, should see similar warning:

![image](https://user-images.githubusercontent.com/7980991/178440243-2ec6fe8b-1cf4-48f3-8cf2-b586f191e4e4.png)


### Checklist

~~-   [ ] There are unit tests to verify my changes are correct or unit tests aren't applicable (if so, write quick reason why unit tests don't exist)~~
~~-   [ ] I've updated relevant documentation in `doc/`.~~
-   [y] I've updated CHANGES.md with what I changed.
-   [y] I've provided instructions in the PR description on how to test this PR.
